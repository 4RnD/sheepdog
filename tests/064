#!/bin/bash

# Test node failure while reweighting
seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_need_to_be_root

_cleanup

_make_device 0 $((200 * 1024 ** 2)) # 200 MB
_make_device 1 $((200 * 1024 ** 2)) # 200 MB
_make_device 2 $((400 * 1024 ** 2)) # 400 MB
_make_device 3 $((200 * 1024 ** 2))  # 200 MB

#start three in different size
for i in 0 1 2; do
	_start_sheep $i
done
_wait_for_sheep 3
$COLLIE cluster format -c 2
sleep 1

$COLLIE vdi create test 100M -P
_node_info
$COLLIE node list

$COLLIE node md plug $STORE/3
_wait_for_sheep_recovery 0
_node_info
$COLLIE node list

$COLLIE cluster reweight

# restart sheep1 while reweighting
_kill_sheep 2
_wait_for_sheep_recovery 0
_node_info | grep -v ^2 # the content of sheep 2 is non-deterministic
$COLLIE node list
_start_sheep 2

_wait_for_sheep_recovery 0
_node_info
$COLLIE node list
$COLLIE cluster info | _filter_cluster_info
