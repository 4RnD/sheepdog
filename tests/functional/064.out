QA output created by 064
using backend plain store
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	12	0	0	0	0	0
1	16	1	0	0	0	0
2	22	1	0	0	0	0
3	0	0	0	0	0	0
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	95          0
   1   127.0.0.1:7001      	95          1
   2   127.0.0.1:7002      	193          2
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	7	0	0	0	0	0
1	16	1	0	0	0	0
2	22	1	0	0	0	0
3	5	0	0	0	0	0
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	95          0
   1   127.0.0.1:7001      	95          1
   2   127.0.0.1:7002      	193          2
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	10	0	0	0	0	0
1	25	1	0	0	0	0
3	15	1	0	0	0	0
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	171          0
   1   127.0.0.1:7001      	85          1
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	8	0	0	0	0	0
1	16	1	0	0	0	0
2	20	1	0	0	0	0
3	6	0	0	0	0	0
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	153          0
   1   127.0.0.1:7001      	76          1
   2   127.0.0.1:7002      	155          2
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      4 [127.0.0.1:7000:153, 127.0.0.1:7001:76, 127.0.0.1:7002:155]
DATE      3 [127.0.0.1:7000:171, 127.0.0.1:7001:85]
DATE      2 [127.0.0.1:7000:153, 127.0.0.1:7001:76, 127.0.0.1:7002:155]
DATE      1 [127.0.0.1:7000:95, 127.0.0.1:7001:95, 127.0.0.1:7002:193]
